build_daemon/lib/client.dart

```diff
@@ -43,10 +43,27 @@ Future<void> _handleDaemonStartup(
 
   // The daemon may log critical information prior to it successfully
   // starting. Capture this data and forward to the logHandler.
+  //
+  // Whenever we see a `logStartMarker` we will parse everything between that
+  // and the `logEndMarker` as a `ServerLog`. Everything else is considered a
+  // normal INFO level log.
+  StringBuffer nextLogRecord;
   var sub = stdout.where((line) => !_isActionMessage(line)).listen((line) {
-    logHandler(ServerLog((b) => b
-      ..level = Level.INFO
-      ..message = line));
+    if (nextLogRecord != null) {
+      if (line == logEndMarker) {
+        logHandler(serializers.deserialize(jsonDecode(nextLogRecord.toString()))
```

Comment:
We probably should put this deserialization in a try catch just in case.

Reply:
done